Source/Authorization/OAuthService.swift

```diff
@@ -87,7 +83,23 @@ class OAuthService {
         }
     }
     
-    private func extractPeAuthorizationCode(from response: PreAuthResponse, completion: @escaping (Result<String, Error>) -> Void) {
+    func requestReferenceToken(oauthToken: OAuthToken, completion: @escaping (Result<TokenSessionResponse, Error>) -> Void) {
+        guard let jwt = JWTUtility.dataTriggerRequestJWT(accessToken: oauthToken.accessToken.value, configuration: configuration) else {
+            fatalError("Invalid reference token request JWT")
+        }
+        
+        apiClient.makeRequest(TokenReferenceRoute(jwt: jwt), completion: completion)
+    }
+    
+    func deleteUser(oauthToken: OAuthToken, completion: @escaping (Result<Void, Error>) -> Void) {
+        guard let jwt = JWTUtility.dataTriggerRequestJWT(accessToken: oauthToken.accessToken.value, configuration: configuration) else {
+            fatalError("Invalid delete user token request JWT")
```

Comment:
We should probably steer away from fatal errors in the context of the SDK, because this will cause the host application to exit.